docs/developers/contributing/performance/profiling.md

```diff
@@ -1,9 +1,19 @@
+(profiling)=
+
 # Profiling
 
-In contrast to performance tracing, profiling does not provide information
-about the timing of events, but instead provides combined information about the
-total time of execution of each function. It also produces a call graph that
-simplifies understanding the call relationship between functions.
+Profiling provides information about the
+total time of execution of each function when performing a specific task or workflow.
+It produces a complete call graph that
+simplifies understanding the call relationship between functions. It is useful for
+identifying which functions are taking most of the runtime and causing bottlenecks.
+
+[Performance monitoring](perfmon), unlike profiling provides information about the
+timing of events or specific functions. It can be useful when profiling identifies a
+common function as taking most of the time in execution, but it's hard to know where
+to optimize because that function gets called from many places.
+It can also help you understand functions that cost a lot but only at the start vs
+functions that are consistently slow — which is hard to track with profiling.
```

Comment:
I was trying to add more information about when perfmon may be useful, relative to profiling. Was from conversation with @jni but this is not fleshed out and I don't fully understand some points. Needs work. Suggestions welcome.

Reply:
Okay updated after more discussion, but happy to make any changes.